fairlearn/metrics/_metric_frame.py

```diff
@@ -414,19 +445,46 @@ def group_min(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+
         Returns
         -------
         typing.Any pandas.Series or pandas.DataFrame
             The minimum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
+        if errors not in _VALID_ERROR_STRING:
+            raise ValueError(_INVALID_ERRORS_VALUE_ERROR_MESSAGE)
+
         if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                min_val = self._by_group[m].min()
-                result[m] = min_val
+            if errors == "raise":
```

Comment:
This looks awfully similar to the same block in `group_min()`, and is now large enough to make me concerned about copying code. Would it be possible to refactor this logic into a separate method, and have the final max/min aggregation be an argument?

Reply:
Yes `group_min `and `group_max `do exactly the same, only .`min()` changes in `.max()`. 

Shall I put all logic in a separate method with parameter `['min', 'max']` which is called from `group_min()` and `group_max()`? Such that the dependencies of `group_min `and `group_max `don't break?

Reply:
Something like that sounds good. But yes, the users of `group_min()` and `group_max()` mustn't break.

Reply:
https://github.com/fairlearn/fairlearn/pull/896/commits/4b6cc6016ac2964fe816986da9244a44ff78c7ea
I fixed it in this way, let me know what you think @riedgar-ms